### ISSUE TITLE
mailboxes: two machines, same setup, different output.

### ISSUE THREAD

**Issue by ahmedelgabri** [neomutt/neomutt#2075]
My mailboxes is setup like this

```
mailboxes "+Work/INBOX" \
  "+Work/Starred" \
  "+Work/Sent" \
  "+Work/Drafts" \
  "+Work/Trash" \
  "+Work/Spam" \
  `tree ~/.mail/Work -d -I "Archive|cur|new|tmp|certs|.notmuch|INBOX|[Gmail]" -afin --noreport | awk '{if(NR>1)print}' | tr '\n' ' '`
```

But the output on both machines are different

Broken behaviour (notice the duplication & also the path expansion `~/.mail/Work/*` instead of `=Work/*`
<img width="1133" alt="broken2" src="https://user-images.githubusercontent.com/63876/72247465-39ba6d00-35f5-11ea-976d-9adce6dbac31.png">

and the sidebar (no hierarchy)
```
 INBOX (1)                   │                                                                                                                      
 Starred (18)                │                                                                                                                     
 Sent (3)                    │
 Drafts                      │
 Trash (29)                  │
 Spam (2)                    │
Drafts                       │
aaaaaaaaaa                   │
bbbbbbbb                     │
Sent (3)                     │
Spam (2)                     │
Starred (18)                 │
Trash (29)                   │
cccccc                       │
company (329)                │
dd                           │
eeee                         │
fffff                        │
ggg                          │
CI (2151)                    │
hhh                          │
iiiiiii (53833)              │
jjjjjjjjjjjjjj               │
github (13343)               │
kkkkkkkkkkkkkk (3)           │
llllll (3299)                │
mmmmmmmm (2494)              │
nnnnnnnnnnnnnn (31103)       │
oooooooo (219)               │
pppp (300)                   │
qqqqqqqqqqqqq                │
jjj (8)                      │
sssssssssssss                │
```

Correct behavior
<img width="569" alt="correct2" src="https://user-images.githubusercontent.com/63876/72247518-548ce180-35f5-11ea-84a9-e957c1fab66a.png">

and the sidebar
```
INBOX (1)                   │
Starred (18)                │
Sent (3)                    │
Drafts                      │
Trash (36)                  │
Spam (2)                    │
aaaaaaaaaa                  │
bbbbbbbb                    │
cccccc                      │
company (329)               │
 dd                         │
 eeee                       │
 fffff                      │
ggg                         │
 CI (446)                   │
 hhh                        │
 iiiiiii (2865)             │
 jjjjjjjjjjjjjj             │
 github (13343)             │
   kkkkkkkkkkkkkk (3)       │
   llllll (149)             │
   mmmmmmmm (146)           │
   nnnnnnnnnnnnnn (3813)    │
   oooooooo (11)            │
  pppppppppppppp (3)        │
  qqqqqq (3694)             │
  rrrrrrrr (3374)           │
  ssssssssssssss (34768)    │
  tttttttt (236)            │
 uuuu (300)                 │
 vvvvvvvvvvvvv              │
 www (8)                    │
xxxxxxxxxxxxx               │
```

Running this 
```
tree ~/.mail/Work -d -I "Archive|cur|new|tmp|certs|.notmuch|INBOX|[Gmail]" -afin --noreport | awk '{if(NR>1)print}' | tr '\n' ' '
``` 
on the command line yields the same exact results on both machines and also both are using the same exact `neomutt` version

```
NeoMutt 20191207
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Darwin 19.2.0 (x86_64)
ncurses: ncurses 5.7.20081102 (compiled with 5.7.20081102)
libiconv: 1.11
libidn: 1.35 (compiled with 1.35)
GPGme: 1.13.1
libnotmuch: 5.2.0
hcache backends: tokyocabinet, lmdb

Configure options: --prefix=/usr/local/Cellar/neomutt/20191207 --enable-gpgme --with-gpgme=/usr/local/opt/gpgme --gss --lmdb --notmuch --sasl --tokyocabinet --with-ssl=/usr/local/opt/openssl@1.1 --with-ui=ncurses

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/local/opt/gpgme/include -I/usr/local/opt/libassuan/include -I/usr/local/opt/libgpg-error/include -I/usr/local/Cellar/neomutt/20191207/include -DNCURSES_WIDECHAR -I/usr/local/opt/openssl@1.1/include

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme +gss +hcache -homespool +idn -inotify 
  -locales_hack -lua +meta -mixmaster +nls +notmuch +openssl +pgp +sasl +smime 
  -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/local/Cellar/neomutt/20191207/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/Cellar/neomutt/20191207/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

**Comment by gahr** [comment]
* Is the rest of your config also identical on both machines? Can you post it (specially `folder`)?
* Is the content of `~/.mail/Work` also identical on both machines? What is that?
* >Running this `tree ~/.mail/Work -d -I "Archive|cur|new|tmp|certs|.notmuch|INBOX|[Gmail]" -afin --noreport | awk '{if(NR>1)print}' | tr '\n' ' '` on the command line yields the same exact results on both machines and also both are using the same exact neomutt version

    I don't have a macOs handy. What's the output?

**Comment by ahmedelgabri** [comment]
> - Is the rest of your config also identical on both machines? Can you post it (specially `folder`)?

This is the work account setup.

```
unmailboxes *

macro index,pager gp "<change-folder>=Personal/INBOX<enter>" "Switch account to Personal"

macro index,pager gs "<change-folder>=Work/Starred<enter>" "go to Starred"
macro browser gs "<exit><change-folder>=Work/Starred<enter>" "go to Starred"
macro index,pager gt "<change-folder>=Work/Sent<enter>" "go to Sent"
macro browser gs "<exit><change-folder>=Work/Sent<enter>" "go to Sent"
macro index,pager gd "<change-folder>=Work/Drafts<enter>" "go to Drafts"
macro browser gs "<exit><change-folder>=Work/Drafts<enter>" "go to Drafts"

macro index,pager / "<vfolder-from-query>path:Work/** " "Searching Work mailbox with notmuch integration in neomutt"

set sendmail = "/usr/local/bin/msmtp -a work"
set from = "foo@bar.com"
set spoolfile = "+Work/INBOX"
set postponed = "+Work/Drafts"
set mbox = "+Work/Archive"
set trash = "+Work/Trash"
set header_cache = "~/.cache/mutt/headers/work/"
set message_cachedir = "~/.cache/mutt/messages/work/"
mailboxes "+Work/INBOX" \
  "+Work/Starred" \
  "+Work/Sent" \
  "+Work/Drafts" \
  "+Work/Trash" \
  "+Work/Spam" \
  `tree ~/.mail/Work -d -I "Archive|cur|new|tmp|certs|.notmuch|INBOX|[Gmail]" -afin --noreport | awk '{if(NR>1)print}' | tr '\n' ' '`

macro index SI "<shell-escape>mbsync work -download<enter>" "sync inbox"
macro index,pager y "<save-message>=Work/Archive<enter>" "Archive conversation"
macro index,pager Y "<tag-thread><save-message>=Work/Archive<enter>" "Archive conversation"

color status red default
color sidebar_highlight black red
color sidebar_indicator red color0
```

Yes, they are & here is [`folder`](https://github.com/ahmedelgabri/dotfiles/blob/fbca618fa4424b837274645689a3d5423f872ef3/files/mail/.mutt/muttrc#L15) in my setup.

> - Is the content of `~/.mail/Work` also identical on both machines? What is that?

Yes, they are identical.

> - > Running this
>   > `tree ~/.mail/Work -d -I "Archive|cur|new|tmp|certs|.notmuch|INBOX|[Gmail]" -afin --noreport | awk '{if(NR>1)print}' | tr '\n' ' '`
>   > on the command line yields the same exact results on both machines and
>   > also both are using the same exact neomutt version
>
>   I don't have a macOs handy. What's the output?

I don't think it's a macOS command only, but here is part of the output the command without the `| tr '\n' ' '` part

Broken machine

```
/Users/ahmed/.mail/Work/Drafts
/Users/ahmed/.mail/Work/aaaaaaaaaa
/Users/ahmed/.mail/Work/bbbbbbbb
/Users/ahmed/.mail/Work/Sent
/Users/ahmed/.mail/Work/Spam
/Users/ahmed/.mail/Work/Starred
/Users/ahmed/.mail/Work/Trash
/Users/ahmed/.mail/Work/cccccc
/Users/ahmed/.mail/Work/company
/Users/ahmed/.mail/Work/company/dd
/Users/ahmed/.mail/Work/company/eeee
/Users/ahmed/.mail/Work/company/fffff
/Users/ahmed/.mail/Work/ggg
/Users/ahmed/.mail/Work/ggg/CI
/Users/ahmed/.mail/Work/ggg/hhh
/Users/ahmed/.mail/Work/ggg/iiiiiii
/Users/ahmed/.mail/Work/ggg/jjjjjjjjjjjjjj
/Users/ahmed/.mail/Work/ggg/github
/Users/ahmed/.mail/Work/ggg/github/kkkkkkkkkkkkkk
/Users/ahmed/.mail/Work/ggg/github/llllll
/Users/ahmed/.mail/Work/ggg/github/mmmmmmmm
/Users/ahmed/.mail/Work/ggg/github/nnnnnnnnn\ nnnn
/Users/ahmed/.mail/Work/ggg/github/oooooooo
/Users/ahmed/.mail/Work/ggg/pppp
/Users/ahmed/.mail/Work/ggg/qqqqqq\ qqqqqq
/Users/ahmed/.mail/Work/ggg/rrr
/Users/ahmed/.mail/Work/sssss\ sssssss
```

Not broken machine

```
/Users/ahmed/.mail/Work/Drafts
/Users/ahmed/.mail/Work/aaaaaaaaaa
/Users/ahmed/.mail/Work/bbbbbbbb
/Users/ahmed/.mail/Work/Sent
/Users/ahmed/.mail/Work/Spam
/Users/ahmed/.mail/Work/Starred
/Users/ahmed/.mail/Work/Trash
/Users/ahmed/.mail/Work/cccccc
/Users/ahmed/.mail/Work/company
/Users/ahmed/.mail/Work/company/dd
/Users/ahmed/.mail/Work/company/eeee
/Users/ahmed/.mail/Work/company/fffff
/Users/ahmed/.mail/Work/ggg
/Users/ahmed/.mail/Work/ggg/CI
/Users/ahmed/.mail/Work/ggg/hhh
/Users/ahmed/.mail/Work/ggg/iiiiiii
/Users/ahmed/.mail/Work/ggg/jjjjjjjjjjjjjj
/Users/ahmed/.mail/Work/ggg/github
/Users/ahmed/.mail/Work/ggg/github/kkkkkkkkkkkkkk
/Users/ahmed/.mail/Work/ggg/github/llllll
/Users/ahmed/.mail/Work/ggg/github/mmmmmmmm
/Users/ahmed/.mail/Work/ggg/github/nnnnnnnnn\ nnnn
/Users/ahmed/.mail/Work/ggg/github/oooooooo
/Users/ahmed/.mail/Work/ggg/pppp
/Users/ahmed/.mail/Work/ggg/qqqqqq\ qqqqqq
/Users/ahmed/.mail/Work/ggg/rrr
/Users/ahmed/.mail/Work/sssss\ sssssss
```

**Comment by ahmedelgabri** [comment]
@gahr I created an exact copy of my mail folder `~/.mail` to `~/mail2` & that one worked fine

A couple of things about `~/.mail` folder:

- It's a symlink to a directory in my dotfiles folder 
- It starts with `.` hidden

These are the only two differences between `~/.mail` & `~/mail2`, although this doesn't answer the question of why on my work machine everything is fine…

**Comment by ahmedelgabri** [comment]
Ok, I fixed the issue & it was related to symlinking (not sure what it is, still)

So, my dotfiles are symlinked using [`GNU Stow`](https://www.gnu.org/software/stow/). When I first install it on a new machine, I create these folders as normal

```
$ mkdir -p ~/.mail/{Work,.notmuch}
```

_Then_ I symlink my config. Recently I changed some folder structure in my dotfiles repo which required me to (un)symlink folders again. And that's when things broke:

- My work machine had the folders not symlinked as above (but pulled the changes from my dotfiles repo) so it worked normally because folders were in place.
- My personal machine had the folders & everything inside of them symlinked instead & that broke `neomutt` `mailboxes` _for some reason_

The fix was simply deleting the symlinks, creating the parent folders as above & symlink config files, etc…
